src/arb/arb.jl

```diff
@@ -85,19 +85,86 @@ characteristic(::ArbField) = 0
 #
 ################################################################################
 
-function Float64(x::arb)
-   GC.@preserve x begin
-      t = ccall((:arb_mid_ptr, libarb), Ptr{arf_struct}, (Ref{arb}, ), x)
-      # 4 == round to nearest
-      d = ccall((:arf_get_d, libarb), Float64, (Ptr{arf_struct}, Int), t, 4)
-   end
-   return d
+@doc Markdown.doc"""
+    Float64(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `Float64`, rounded in the direction specified by $round$.
+For `RoundNearest` the return value approximates the midpoint of $x$. For
+`RoundDown` or `RoundUp` the return value is a lower bound or upper bound for
+all values in $x$.
+"""
+function Float64(x::arb, round::RoundingMode=RoundNearest)
+  t = _arb_get_arf(x, round)
+  return _arf_get_d(t, round)
+end
+
+@doc Markdown.doc"""
+    BigFloat(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `BigFloat` of the currently used precision, rounded in the
+direction specified by $round$. For `RoundNearest` the return value
+approximates the midpoint of $x$. For `RoundDown` or `RoundUp` the return
+value is a lower bound or upper bound for all values in $x$.
+"""
+function BigFloat(x::arb, round::RoundingMode=RoundNearest)
```

Comment:
`BigFloat` constructors usually respect the global precision (sorry):
```julia
julia> setprecision(20) do
         BigFloat("1.000000000000001")
       end
1.0
```
I am not sure we should do it here. What do you think?

Reply:
Accommodations can be made. Do we have any BigFloat constructors here that don't respect Julia's design decisions?

Reply:
I just tried `fmpz` and it respect the global BigFloat precision (since it just does `BigFloat(BigInt(x))`. Don't think we have other conversion functions.

Reply:
Ya gotta get to the point. Where in this code is a BigFloat created with the wrong precision?